compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -1045,6 +1047,63 @@ static unsigned decideFusableLinalgOps(FunctionOpInterface funcOp,
   return numRootOps;
 }
 
+FusionGroupMapping decideFusionGroups(FunctionOpInterface funcOp,
+                                      DominanceInfo const &dominanceInfo) {
+  unsigned numRoots = decideFusableLinalgOps(funcOp, dominanceInfo);
+  SmallVector<Operation *> roots(numRoots, nullptr);
+
+  // Gather all roots.
+  funcOp.walk([&](Operation *op) {
+    if (hasRootOpAttribute(op)) {
+      roots[getRootNumber(op)] = op;
+      removeRootOpAttribute(op);
+    }
+  });
+
+  // Gather tileable producers.
+  FusionGroupMapping result;
```

Comment:
I am not familiar enough with the code below, please wait until @MaheshRavishankar has made a pass.